ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorDelegate.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.execution;
+
+import io.confluent.ksql.api.plugin.PullQueryApiExecutor;
+import io.confluent.ksql.api.plugin.TableRows;
+import io.confluent.ksql.parser.tree.Query;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.rest.entity.TableRowsEntity;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import java.util.List;
+
+/**
+ * Wrapper to break circular dependencies between modules. This is temporary until we migrate all
```

Comment:
To check my understanding: the problem with circular dependencies in this PR is that the new ksql-api module wants access to `PullQueryExecutor` and related files from `ksql-rest-app`, but that's not possible right now because `ksql-rest-app` depends on `ksql-api`, yes? And the reason you say this is temporary is because once the new API work is done, `ksql-api` and `ksql-rest-app` will be unified into a single module? Or did you have other plans

Reply:
Specifically, we need to call PullQueryExecutor from ksql-api, but that returns TableRowsEntity which is in ksql-rest-model. ksql-api can't depend on ksql-rest-model, as ksql-rest-app depends on ksql-rest-model and ksql-api. If it did depend on ksql-rest-model there would be a circular dep.

Reply:
This is temporary as ksql-rest-app and ksql-rest-model will disappear when KLIP-15 is complete.